ledger/ledger.go

```diff
@@ -647,9 +653,54 @@ func (l *Ledger) VerifiedTransactionCache() verify.VerifiedTransactionCache {
 	return l.verifiedTxnCache
 }
 
-// TxLease is an exported version of txlease
-type TxLease struct {
-	ledgercore.Txlease
+// StartEvaluator creates a BlockEvaluator, given a ledger and a block header
+// of the block that the caller is planning to evaluate. If the length of the
+// payset being evaluated is known in advance, a paysetHint >= 0 can be
+// passed, avoiding unnecessary payset slice growth.
+func (l *Ledger) StartEvaluator(hdr bookkeeping.BlockHeader, paysetHint int) (*internal.BlockEvaluator, error) {
+	proto, ok := config.Consensus[hdr.CurrentProtocol]
+	if !ok {
+		return nil, protocol.Error(hdr.CurrentProtocol)
+	}
+
+	return internal.StartEvaluator(l, hdr, proto, paysetHint, true, true)
+}
+
+// Validate uses the ledger to validate block blk as a candidate next block.
+// It returns an error if blk is not the expected next block, or if blk is
+// not a valid block (e.g., it has duplicate transactions, overspends some
+// account, etc).
+func (l *Ledger) Validate(ctx context.Context, blk bookkeeping.Block, executionPool execpool.BacklogPool) (*ledgercore.ValidatedBlock, error) {
+	delta, err := internal.Eval(ctx, l, blk, true, l.verifiedTxnCache, executionPool)
+	if err != nil {
+		return nil, err
+	}
+
+	vb := ledgercore.MakeValidatedBlock(blk, delta)
+	return &vb, nil
+}
+
+// CompactCertParams computes the parameters for building or verifying
+// a compact cert for block hdr, using voters from block votersHdr.
+func CompactCertParams(votersHdr bookkeeping.BlockHeader, hdr bookkeeping.BlockHeader) (res compactcert.Params, err error) {
+	return internal.CompactCertParams(votersHdr, hdr)
+}
+
+// AcceptableCompactCertWeight computes the acceptable signed weight
+// of a compact cert if it were to appear in a transaction with a
+// particular firstValid round.  Earlier rounds require a smaller cert.
+// votersHdr specifies the block that contains the Merkle commitment of
+// the voters for this compact cert (and thus the compact cert is for
+// votersHdr.Round() + CompactCertRounds).
+//
+// logger must not be nil; use at least logging.Base()
+func AcceptableCompactCertWeight(votersHdr bookkeeping.BlockHeader, firstValid basics.Round, logger logging.Logger) uint64 {
+	return internal.AcceptableCompactCertWeight(votersHdr, firstValid, logger)
+}
+
+// MakeDebugBalances creates a ledger suitable for dryrun and debugger
+func MakeDebugBalances(l ledgercore.LedgerForCowBase, round basics.Round, proto protocol.ConsensusVersion, prevTimestamp int64) apply.Balances {
```

Comment:
maybe make a debugging interface with aliasing like this?
```
type LedgerForDebugger = internal.LedgerForCowBase
```
In this case `LedgerForCowBase` can go to `internal` where it should live

Reply:
thoughts/comments?